c/zss.c

```diff
@@ -319,10 +321,76 @@ static void setPrivilegedServerName(HttpServer *server, JsonObject *mvdSettings,
 }
 #endif /* __ZOWE_OS_ZOS */
 
+typedef struct RbacAuthorizationData_t {
+  int instanceId;
+} RbacAuthorizationData;
+
+static int rbacAuthorization(HttpService *service, HttpRequest *request, HttpResponse *response, void *userData) {
+  if (request->username == NULL) {
+    // username is required to perform RBAC check
+    // would a check for (service->authType != SERVICE_AUTH_NATIVE_WITH_SESSION_TOKEN) be better?
+    return TRUE;
+  }
+
+  RbacAuthorizationData *rbacData = userData;
+
+  char method[16];
+  snprintf(method, sizeof(method), "%s", request->method);
+  destructivelyNativize(method);
+
+  char profileName[ZOWE_PROFILE_NAME_LEN+1] = {0};
+  int rc = getProfileNameFromRequest(profileName, sizeof(profileName), request->parsedFile, method, rbacData->instanceId);
+  if (rc != 0) {
+    return FALSE;
+  }
+
+  rc = verifyAccessToSafProfile(service->server, request->username, profileName, SAF_AUTH_ATTR_READ);
+  if (rc != 0) {
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+static bool isRbacEnabled(JsonObject *mvdSettings, JsonObject *envSettings) {
+  int rbacParm = FALSE;
+  Json *rbacObj = jsonObjectGetPropertyValue(envSettings, "ZWED_dataserviceAuthentication_rbac");
+  if (rbacObj == NULL) {
+    JsonObject *dataserviceAuth = jsonObjectGetObject(mvdSettings, "dataserviceAuthentication");
+    if (dataserviceAuth) {
+      rbacParm = jsonObjectGetBoolean(dataserviceAuth, "rbac");
+    }
+  } else {
+    rbacParm = jsonAsBoolean(rbacObj);
+  }
+  return rbacParm;
+}
+
+static int getZoweInstanceId() {
```

Comment:
Declare functions that take no parameters using a `void` argument, for example:
```c
int foo(void);
```
Otherwise it's considered "unspecified number of parameters" by the standard.